src/istgt_iscsi.c

```diff
@@ -5343,7 +5363,7 @@ wait_all_task(CONN_Ptr conn)
 		conn->id, conn->running_tasks);
 }
 
-
+#if 0
```

Comment:
These changes seem unrelated.

Reply:
This piece of thread cleanup code is trying to take care of unexpected thread exits and is buggy. However, unexpected thread exits are not handled in the entire code, and, thus this code need to be removed.
This is related to current issue as this code uses lot of locks. These locks can impact the processing of SendTargets requests. So, removed this code in this PR.

Reply:
Ok that makes sense. Im in favor of removing all the locks =) BUT. As this is a workaround, and we want to revert his in the near future, it will revert all. But if that's OK then I'm fine with it.

Reply:
while reverting the changes, we need to do with care :)